w3af/core/data/parsers/doc/open_api/main.py

```diff
@@ -178,9 +180,65 @@ def parse(self):
                 if not self._should_audit(fuzzable_request):
                     continue
 
+                if self.discover_fuzzable_headers:
+                    operation = data[4]
+                    headers = self._get_parameter_headers(operation)
+                    fuzzable_request.set_force_fuzzing_headers(headers)
+
                 self.api_calls.append(fuzzable_request)
 
-    def _should_audit(self, fuzzable_request):
+    def _get_parameter_headers(self, operation):
+        """
+        Looks for all parameters which are passed to the endpoint via headers.
+        The method filters out headers with auth info
+        which are defined in 'securityDefinitions' section of the API spec.
+
+        :param operation: An instance of Operation class
+                          which represents the API endpoint.
+        :return: A list of unique header names.
+        """
+        parameter_headers = set()
+        for parameter_name, parameter in operation.params.iteritems():
+            if parameter.location == 'header':
+
+                # Fuzz auth headers only once.
+                if self._is_already_fuzzed_auth_header(parameter.name, operation.swagger_spec):
```

Comment:
The parser shouldn't have any responsibility associated with fuzzing. The parser should just return what is found in the OpenAPI, then other parts of the code should be responsible of fuzzing (or not) the authentication header.

Not sure where to put this... maybe the solution is to modify the [fuzzer.py](https://github.com/andresriancho/w3af/blob/master/w3af/core/data/fuzzer/fuzzer.py#L35-L60) code?